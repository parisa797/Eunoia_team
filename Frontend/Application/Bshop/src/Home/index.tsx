import * as React from "react";
import { createDrawerNavigator } from "@react-navigation/drawer";

import { HomeRoutes } from "../components/Navigation";
import DrawerContent, { DRAWER_WIDTH } from "./Drawer";
export { assets } from "./Drawer";

import OutfitIdeas from "./OutfitIdeas/OutfitIdeas";
import EditProfile from "./EditProfile";

const Drawer = createDrawerNavigator<HomeRoutes>();
export const HomeNavigator = () => (
  <Drawer.Navigator
    drawerContent={() => <DrawerContent />}
    drawerStyle={{ width: DRAWER_WIDTH }}
  >
    <Drawer.Screen name="OutfitIdeas" component={OutfitIdeas} />
    <Drawer.Screen name="EditProfile" component={EditProfile} />
   
  </Drawer.Navigator>
);
