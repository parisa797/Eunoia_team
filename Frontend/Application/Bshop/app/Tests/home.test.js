import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitFor,
} from "react-native-testing-library";
import renderer from "react-test-renderer";
// import { shallow } from "enzyme";

import Home from "../homepage";

describe("Home page with shops be tested", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it("fetch test", async () => {
  //   const shops = [
  //     {
  //       title: "shop1",
  //       rate_value: 0,
  //       rate_count: 10,
  //       id: 0,
  //       address: "ادرس ادرس آدرسسسسس",
  //       logo: "sljfleij.png",
  //     },
  //     {
  //       title: "shop2-سینمتبمث",
  //       rate_value: 1.23482394,
  //       id: 1,
  //       address: "",
  //       logo: "",
  //     },
  //     {
  //       title: "شاپ3",
  //       rate_value: 2.3,
  //       rate_count: 11,
  //       id: 2,
  //       address: "a",
  //       logo: null,
  //     },
  //     {
  //       title: "",
  //       rate_value: 5,
  //       id: 3,
  //       address: "addressssss sdfe",
  //       logo: "",
  //     },
  //   ];
  //   jest.spyOn(global, "fetch").mockImplementation(() =>
  //     Promise.resolve({
  //       status: 200,
  //       json: () => shops,
  //     })
  //   );

  //   const push = jest.fn();
  //   const { getByTestId, getByPlaceholderText } = await render(
  //     <Home navigation={{ push }} />
  //   );
  //   await new Promise((resolve) => setImmediate(resolve));
  // });

  //   it("fetches data from server when server returns a successful response", (done) => {
  //     // 1
  //     const mockSuccessResponse = {};
  //     const mockJsonPromise = Promise.resolve(mockSuccessResponse); // 2
  //     const mockFetchPromise = Promise.resolve({
  //       // 3
  //       json: () => mockJsonPromise,
  //     });
  //     // jest.spyOn(global, "fetch").mockImplementation(() => mockFetchPromise); // 4
  //     global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);

  //     const wrapper = shallow(<Home />); // 5

  //     expect(global.fetch).toHaveBeenCalledTimes(1);
  //     expect(global.fetch).toHaveBeenCalledWith(
  //       "http://iust-bshop.herokuapp.com/api/v1/shops/"
  //     );

  //     process.nextTick(() => {
  //       // 6
  //       //   expect(wrapper.state()).toEqual({
  //       //     // ... assert the set state
  //       //   });

  //       global.fetch.mockClear(); // 7
  //       done(); // 8
  //     });
  //   });

  //   it("fetch test", async () => {
  //     const push = jest.fn();
  //     const { getByTestId, getByPlaceholderText } = await render(
  //       <Home navigation={{ push }} />
  //     );

  //     const shop = [
  //       {
  //         title: "فروشگاه",
  //         phone: "09900000000",
  //         id: 1,
  //       },
  //     ];

  //     const mocked = jest.spyOn(global, "fetch").mockImplementation(() =>
  //       Promise.resolve({
  //         status: 200,
  //         json: () => shop,
  //       })
  //     );

  //     expect(mocked).toHaveBeenCalled();
  //   });
});
