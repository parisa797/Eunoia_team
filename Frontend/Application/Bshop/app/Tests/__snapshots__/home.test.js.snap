// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page with shops be tested renders correctly 1`] = `
<RCTScrollView
  nestedScrollEnabled={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "marginBottom": "5%",
      "marginTop": "5%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#f1f1f2",
          "borderRadius": 10,
          "height": 45,
          "marginBottom": "5%",
          "marginTop": "10%",
          "width": "90%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        placeholder="جستجو"
        placeholderTextColor="#000"
        rejectResponderTermination={true}
        style={
          Object {
            "flex": 1,
            "fontSize": 20,
            "height": 50,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      nestedScrollEnabled={true}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshing={false}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTScrollView>
`;
